pkg/client/unversioned/client.go

```diff
@@ -171,6 +175,49 @@ func (c *Client) ValidateComponents() (*api.ComponentStatusList, error) {
 	return &api.ComponentStatusList{Items: statuses}, nil
 }
 
+// SwaggerSchemaInterface has a method to retrieve the swagger schema. Used in
+// client.Interface
+type SwaggerSchemaInterface interface {
+	SwaggerSchema(version string) (*swagger.ApiDeclaration, error)
+}
+
+// SwaggerSchema retrieves and parses the swagger API schema the server supports.
+func (c *Client) SwaggerSchema(version string) (*swagger.ApiDeclaration, error) {
+	if version == "" {
+		version = latest.Version
+	}
+
+	vers, err := c.ServerAPIVersions()
+	if err != nil {
+		return nil, err
+	}
+
+	// This check also takes care the case that kubectl is newer than the running endpoint
+	if stringDoesntExistIn(version, vers.Versions) {
+		return nil, fmt.Errorf("API version: %s is not supported by the server. Use one of: %v", version, vers.Versions)
```

Comment:
What if users dont specify the apiversion, for ex: ```kubectl doc Pod``` do we use the latest version by default (like print v1.Pod, in this case) or are users required to write ```kubectl doc v1.Pod```
We could also support ```kubectl doc Pod --api-version=v2```, since api-version is expected to work across all commands.